### PR TITLE
Add TryGet/ServiceEnable/ServiceDisable

### PR DESCRIPTION
**Pull request by afarchy**

### CODE REVIEW

**Review by NicholasPerell — APPROVED** [review]
I like the addition of the _servicesInOrder list for the Initialize and Shutdown. Really cool.

File: ServiceLocator/Packages/com.virtualmaker.servicelocator/Runtime/ServiceLocator.cs [hunk 1]
```diff
@@ -50,31 +57,68 @@ private static ServiceLocator GetOrCreate()
...
-            return (T)service;
+            service = (T)svc;
+            return true;
+        }
+
+        private void OnEnable()
+        {
+            foreach (var service in _servicesInOrder)
+            {
+                service.Initialize();
```
Comment (StephenHodgson) [inline ServiceLocator/Packages/com.virtualmaker.servicelocator/Runtime/ServiceLocator.cs]: kinda wish these had the same name as OnEnable/OnDisable so they're more obvious about where in the lifecycle of events they are called form.

File: ServiceLocator/Packages/com.virtualmaker.servicelocator/Runtime/ServiceLocator.cs [hunk 2]
```diff
@@ -50,31 +57,68 @@ private static ServiceLocator GetOrCreate()
...
             {
-                throw new Exception("ServiceLocator not found");
+                // instance can be null if called while stopping play mode
+                // because ServiceLocator is destroyed before other objects.
+                service = default!;
+                return false;
             }
 
-            if (!instance._services.TryGetValue(typeof(T), out var service))
+            if (!instance._services.TryGetValue(typeof(T), out var svc))
```
Comment (StephenHodgson) [inline ServiceLocator/Packages/com.virtualmaker.servicelocator/Runtime/ServiceLocator.cs]:
```suggestion
            if (!instance._services.TryGetValue(typeof(T), out var knownService))
```

**Review by StephenHodgson — APPROVED** [review]
looks good. Seems more in line with the [original spec](https://github.com/XRTK/com.xrtk.core/blob/8fe0d49189e985617304169d96be2e5226872633/XRTK-Core/Packages/com.xrtk.core/Runtime/Services/MixedRealityToolkit.cs#L369) from years ago

File: ServiceLocator/Packages/com.virtualmaker.servicelocator/Runtime/ServiceLocator.cs [hunk 3]
```diff
@@ -101,23 +102,23 @@ private static ServiceLocator GetOrCreate()
...
             {
-                service.Initialize();
+                service.ServiceEnable();
             }
         }
 
         private void OnDisable()
         {
-            foreach (var service in _servicesInOrder)
+            for (int i = _servicesInOrder.Count - 1; i >= 0; i--)
```
Comment (StephenHodgson) [inline ServiceLocator/Packages/com.virtualmaker.servicelocator/Runtime/ServiceLocator.cs:134]: good catch

**Review by StephenHodgson — APPROVED** [review]

**Pull request merged**